pkg/parse/run.go

```diff
@@ -120,6 +120,13 @@ func DefaultRunFunc(ctx context.Context, r Reconciler, trigger string) RunResult
 	if gs.Errs == nil {
 		if err := r.SyncStatusClient().SetImageToSyncAnnotation(ctx, gs.Commit); err != nil {
```

Comment:
A comment that is not specific to the unblocking feature. Now that we have `ReconcilerState.ReconcilerStatus` populated from RSync, we can skip setting the annotation and updating the ready-to-render file when the commit is not changed.

In `type ReconcilerStatus struct {}`, we can add a field `ImageToSyncAnnotation` to cache the annotation, and here we update `if gs.Errs == nil {` to `if gs.Errs == nil && r.ReconcilerState().status.ImageToSyncAnnotation != gs.Commit {`.

This can be a separate PR.